stdlib/universe/limit.go

```diff
@@ -238,3 +246,110 @@ func (t *limitTransformation) UpdateProcessingTime(id execute.DatasetID, pt exec
 func (t *limitTransformation) Finish(id execute.DatasetID, err error) {
 	t.d.Finish(err)
 }
+
+type limitState struct {
+	n      int
+	offset int
+}
+type limitTransformationAdapter struct {
+	limitTransformation *limitTransformation
+}
+
+func (*limitTransformationAdapter) Close() error {
+	return nil
+}
+
+func (t *limitTransformationAdapter) Process(
+	chunk table.Chunk,
+	state interface{},
+	dataset *execute.TransportDataset,
+	_ arrowmem.Allocator,
+) (interface{}, bool, error) {
+
+	var state_ *limitState
+	// `.Process` is reentrant, so to speak. The first invocation will not
+	// include a value for `state`. Initialization happens here then is passed
+	// in/out for the subsequent calls.
+	if state == nil {
+		state_ = &limitState{n: t.limitTransformation.n, offset: t.limitTransformation.offset}
+		// When state initialization is happening, we know this is the first
+		// invocation of `.Process` for this transformation.
+		// When the incoming chunk is empty we can short-circuit,
+		// passing along the empty, skipping further processing.
+		if chunk.Len() == 0 {
+			chunk.Retain()
+			if err := dataset.Process(chunk); err != nil {
+				return nil, false, err
+			}
+			return nil, true, nil
+		}
```

Comment:
Looking for someone to tell me if this is true :cold_sweat:. My understanding is the transform will not receive empty chunks sprinkled throughout the stream and if we start empty, things will stay empty. Is that right?